tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -437,13 +437,16 @@ def get_use_or_def_segment_at_position(
         a definition when it can, and falls back to references otherwise.
         """
 
-        segment = self.get_def_segment_at_position(position)
-        if segment:
-            return segment
-        else:
-            segment = self.get_use_segment_at_position(position)
-            if segment:
-                return segment.resolved_to
+        segment_def = self.get_def_segment_at_position(position)
+        segment_use = self.get_use_segment_at_position(position)
+
+        if segment_def and not segment_use:
+            return segment_def
+        elif segment_use and not segment_def:
+            return segment_use.resolved_to
+        elif segment_use and segment_def:
+            # prefer the use
+            return segment_use.resolved_to
```

Comment:
I don't think you should make this schange. In my opinionm the if-else structure already obviously highlights what gets preferred when (I see the code before preferred def; I'm fine with preferring use). Moreover, unlike the original version, this now performs two searches instead of one in the case where a use is found: there's redundancy. I strongly believe you should leave the structure as-is, and swap the calls if you want to prioritize `use`:

```Python
        segment = self.get_use_segment_at_position(position)
        if segment:
            return segment.resolved_to
        else:
            segment = self.get_def_segment_at_position(position)
            if segment:
                return segment
```

Reply:
Use does need to be preferred because in the case that they overlap.

```chapel
var a = b;
```

If a user highlights/selects/hovers on `b`, because the def is preferred they get `a`. But if a user hovers over `a`, they should get `a`.  

I will adjust to the previous structure with the preference for `use`

Reply:
```Chapel
var a = b;
```
Is definitely a good example of why we need `use` first.